src/transformers/models/whisper/modeling_whisper.py

```diff
@@ -996,7 +996,9 @@ def forward(
         )
 
         # embed positions
-        positions = self.embed_positions(input_ids, past_key_values_length=past_key_values_length)
+        positions = self.embed_positions(input_ids, past_key_values_length=past_key_values_length).to(
+            inputs_embeds.device
+        )
```

Comment:
Hope this is the way to go. I feel a bit uncertain to move this around due to the offload mechanism.